### PR TITLE
Pact testing docs

### PR DESCRIPTION
**Pull request by blackbaud-johnly**

### CODE REVIEW

File: src/app/learn/reference/configuration/index.html [hunk 1]
```diff
@@ -161,6 +161,56 @@
...
+          </p>
+        </li>
+        <li>
+          <p>
+            <stache-code>host</stache-code> — Specifies a host for the pact server. By default, the host is <stache-code>localhost</stache-code>. The other valid value for this setting is the base URL to communicate with SKY UX Host, which you specify in the <stache-code>url</stache-code> setting of the <stache-code>host</stache-code> property in <stache-code>skyuxconfig.json</stache-code>. (The <stache-code>url</stache-code> setting defaults to <stache-code>https://host.nxt.blackbaud.com</stache-code>.) The <stache-code>skyux pact</stache-code> command uses the <stache-code>pacts</stache-code> property's <stache-code>host</stache-code> value for CORS to ensure that mocked requests work like normal requests. The proxy server sets the <stache-code>Origin</stache-code> header on the request to the <stache-code>host</stache-code> value.
```
[the remark this answers is not in the document]
Comment (blackbaud-joshlandi) [reply src/app/learn/reference/configuration/index.html]: Ah, I see it's in the tutorial.  Well maybe then take this information out and leave it to the tutorial to explain how `host.url` is used.

File: src/app/learn/get-started/advanced/pact-tests/index.html [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                  'Content-Type': {expected content type},
+                  'Access-Control-Allow-Origin': 'host.nxt.blackbaud.com' (or whatever your SPA domain is)
+                }
+              }
+            }
+          </stache-code-block>
+        </li>
+        <li>
+          <p>
+            <stache-code>removeInteractions(provider)</stache-code> — Removes all interactions from server. This is useful to start a new test file in a suite of tests to make sure no clash between occurs expected interactions.
```
Comment (blackbaud-joshlandi) [inline src/app/learn/get-started/advanced/pact-tests/index.html]: ` no clash between occurs` should be `no clash occurs between`

**Comment by blackbaud-joshlandi** [comment]
Also worth noting is that the pact command doesn't come up on `https://host.nxt.blackbaud.com/skyux/learn/reference/cli-commands`

File: src/app/learn/reference/configuration/index.html [hunk 3]
```diff
@@ -161,6 +161,56 @@
...
+          </p>
+        </li>
+        <li>
+          <p>
+            <stache-code>dir</stache-code> — Specifies a directory for the pact file that the pact server generates when <stache-code>skyux pact</stache-code> runs. By default, the pact server stores the JSON file in the <stache-code>app/pacts</stache-code> directory. The file includes an array of request/response pairs called interactions to describe the behaviors that the consumer expects from the provider service. The file name follows the format of <stache-code>&#123;consumer}-&#123;provider}.json</stache-code>, where <stache-code>&#123;consumer}</stache-code> is the name of the consumer and <stache-code>&#123;provider}</stache-code> is the name of the provider. The pact file needs to be uploaded to the pact broker, which for Blackbaud developers is at <a href="https://host.nxt.blackbaud.com/pact/">https://host.nxt.blackbaud.com/pact/</a>. This internal Blackbaud-only site is not available to external developers.
```
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/learn/reference/configuration/index.html]: I'll change:
"The pact file needs to be uploaded to the pact broker, which for Blackbaud developers is at <a href="https://host.nxt.blackbaud.com/pact/">https://host.nxt.blackbaud.com/pact/</a>. This internal Blackbaud-only site is not available to external developers."

To:
"The pact file needs to be uploaded to the pact broker. Blackbaud developers can find information on registered services at <a href="https://host.nxt.blackbaud.com/pact/">https://host.nxt.blackbaud.com/pact/</a>. This internal Blackbaud-only site is not available to external developers."

Let me know if that doesn't work.

File: src/app/learn/reference/configuration/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/learn/reference/configuration/index.html]: I guess I didn't understand the explanations for this property. Sorry about that. I'll trim this down to:
"<stache-code>host</stache-code> — Specifies a host for the pact server. By default, the host is <stache-code>localhost</stache-code>."

Anything else that you think we should call out about this property?

File: src/app/learn/reference/cli-commands/pact/index.html [hunk 4]
```diff
@@ -1,8 +1,16 @@
...
+  <p>
+    <stache-code>skyux pact</stache-code> runs pact tests. Pact is a testing framework that you can use to test a SPA's integration with the services that it consumes. Karma is configured to run any files in the <stache-code>src/app</stache-code> folder with names that end in <stache-code>.pact-spec.ts</stache-code>. The <stache-code>pact</stache-code> command reads the <stache-code>pacts</stache-code> setting in the <stache-code>skyuxconfig.json</stache-code> configuration file and spins up an individual pact server for each provider service specified in the <stache-code>pacts</stache-code> property. It also spins up a pact server for the proxy server. We recommend creating pact tests for all services that the SPA consumes to confirm that your SPA can call out to the provider services and vice versa. For information about how to configure the SPA for pact testing, see <a routerLink="/learn/reference/configuration">the configuration reference</a>. And for more information on configuration settings and guidance on how to write pact tests, see <a routerLink="/learn/get-started/advanced/pact-tests">the pact testing tutorial</a>.
```
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/learn/reference/cli-commands/pact/index.html]: I've expanded the cli commands explanation for the proxy server to include this note as well as some additional content.

File: src/app/learn/get-started/advanced/pact-tests/index.html [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+              "host": "{host}",
+              "port": "{port}"
+            }
+          ]
+        }
+      </stache-code-block>
+    </li>
+    <li>
+      <p>
+        To specify a directory for the pact file that the pact server generates to describe the behaviors that the consumer expects from the provider service, add the <stache-code>log</stache-code> setting. By default, the pact server stores the JSON file in the <stache-code>app/pacts</stache-code> directory.
```
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/learn/get-started/advanced/pact-tests/index.html]: Copy and paste error. Thanks for catching that.

File: src/app/learn/get-started/advanced/pact-tests/index.html [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+        }
+        </stache-code-block>
+    <li>
+      <p>
+        Configure the log file as necessary:
+      </p>
+      <ul>
+        <li>
+          <p>
+            To specify a location for the log file that includes debug output to help diagnose issues, add the <stache-code>dir</stache-code> setting. By default, the file is placed in the <stache-code>app/pacts</stache-code> directory.</p>
```
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/learn/get-started/advanced/pact-tests/index.html]: Copy and paste errors. Thanks again.

**Comment by blackbaud-johnly** [comment]
Thanks for the feedback, @blackbaud-joshlandi! I think I've addressed everything you called out. If you could give the docs another look when you get a chance, particularly the expanded topic for the `pact` command and the shorter entry for the `host` property, I'd really appreciate it. Thanks again.

**Pull request merged**